include/dict0types.h

```diff
@@ -530,6 +534,50 @@ struct dict_table_t {
   IF_DEBUG(ulint magic_n;)
 };
 
+class dict_table_lru {
+ private:
+  using table_list_t = std::list<dict_table_t *>;
```

Comment:
You are better off naming this Tables and the Iterator  as Iter. It shouldn't be important what the underlying data structure is.